app/controllers/topics_controller.rb

```diff
@@ -1,7 +1,8 @@
 class TopicsController < ApplicationController
 
   before_action :require_sign_in, except: [:index, :show]
-  before_action :authorize_user, except: [:index, :show]
+  before_action :authorize_user, except:  [:index, :show]
+  before_action :moderator_check, except: [:index, :show,:edit, :update]
```

Comment:
Actually I take it back.. but I would recommend you rename these methods.  I expected `moderator_check` to check if the user is a moderator.